ir/shared/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -770,18 +770,24 @@ object Serializers {
         buffer.writeInt(encodedNameToIndex(originalName.get))
     }
 
-    def writeParamDef(paramDef: ParamDef): Unit = {
+    def writeParamDef(paramDef: ParamDef, rest: Boolean = false): Unit = {
       writePosition(paramDef.pos)
       writeLocalIdent(paramDef.name)
       writeOriginalName(paramDef.originalName)
       writeType(paramDef.ptpe)
       buffer.writeBoolean(paramDef.mutable)
-      buffer.writeBoolean(paramDef.rest)
+      buffer.writeBoolean(rest)
     }
 
-    def writeParamDefs(paramDefs: List[ParamDef]): Unit = {
-      buffer.writeInt(paramDefs.size)
-      paramDefs.foreach(writeParamDef)
+    def writeParamDefs(paramDefs: List[ParamDef],
+        restParamDef: Option[ParamDef] = None): Unit = {
```

Comment:
Consider adding a `// temp` comment telling that this tests backward compat. I was quite confused for a while that this method implemented the old scheme whereas the reading would be trying to read the new scheme.

Same in `writeParamDef`, I guess.